### ISSUE TITLE
confused about fine tuning ....

### ISSUE THREAD

**Issue by VanitarNordic** [DrSleep/tensorflow-deeplab-resnet#72]
Hello,

I got confused about to how to fine-tune the model by the previous issues. may I ask you to clear it up once forever?
Number of classes will vary, more or less.

**Comment by DrSleep** [comment]
It is better if you provide the details on where you got confused.
Check out new instructions first here: https://github.com/DrSleep/tensorflow-deeplab-resnet#using-your-dataset

**Comment by VanitarNordic** [comment]
@DrSleep 

Your new updated readme helped a lot. Thank you a lot.

 just these questions:

> Change the IMG_MEAN vector accordingly in the script file that you will be using

1) Would you please explain this section more?


> At the moment, the post-processing step with CRF is not implemented. Besides that, multi-scale inputs are missing, as well. No weight regularisation is applied.

2) How much this would affect the performance?

**Comment by DrSleep** [comment]
1. As a pre-processing step for normalisation, we extract mean vector of
RGB values from each image. The values of this vector depend on your
dataset. So if you use other than Pascal dataset, you may need to change
the vector. But most of the time, the one used for Pascal will work for
other datasets.

2. The multi-scale inputs and regularisation are already implemented. CRF
usually gives about ~1% improvement. You can check out the DeepLab paper
for the exact number

**Comment by VanitarNordic** [comment]
@DrSleep 

Thank you again. 

Where is this `IMG_MEAN vector`? I mean in which file? as you know, in Caffe, there is an option to calculate it when we start to train actually (subtract mean: image/pixel).

**Comment by DrSleep** [comment]
"Change the IMG_MEAN vector accordingly in the script file that you will be
using..."
It is in train, train_msc,  evaluate,  etc.

**Comment by VanitarNordic** [comment]
Thank you again.

I have these questions also.

1) We should not use `fine_tune.py` to fine tune the model? Why you have referred to `train.py`?

2) Where should we include the pretrained model (I suppose should be a `.caffemodel` file which was converted to tensorflow)?
I suppose we should change this line: `RESTORE_FROM = './deeplab_resnet.ckpt'`

**Comment by DrSleep** [comment]
1. If you want to fine tune (train only last layers), use fine_tune.py.
Sometimes you want to train the whole network: in that case, use train.py
or train_msc.py.
2. Yes.

**Comment by VanitarNordic** [comment]
May I ask you where should we include the `val.txt` (validation images)?

Here I only see the `train.txt`

```
BATCH_SIZE = 4
DATA_DIRECTORY = '/home/VOCdevkit'
DATA_LIST_PATH = './dataset/train.txt'
IGNORE_LABEL = 255
INPUT_SIZE = '321,321'
LEARNING_RATE = 1e-4
NUM_CLASSES = 21
NUM_STEPS = 20000
RANDOM_SEED = 1234
RESTORE_FROM = './deeplab_resnet.ckpt'
SAVE_NUM_IMAGES = 2
SAVE_PRED_EVERY = 100
SNAPSHOT_DIR = './snapshots_finetune/'
```

**Comment by VanitarNordic** [comment]
if I am right, the training and validation perform by separate commands: `train` and `evaluate`. 

it is not possible to check the accuracy after each epoch or iteration to halt the training at early exist point, I mean before over-fitting?

**Comment by DrSleep** [comment]
train.py for training and evaluate.py for evaluation. You can add another
dataset yourself in the training script to monitor any metrics you want.

**Comment by VanitarNordic** [comment]
Thanks.

I want to use Tensorflow in Windows. Have you tested the repo in windows? from the paths I see that it is tested on Linux.

**Comment by DrSleep** [comment]
Have not tried windows
